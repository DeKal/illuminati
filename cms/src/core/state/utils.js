export const toSuccess = actionName => `${actionName}Success`

export const toError = actionName => `${actionName}Error`

export const toRequest = actionName => `${actionName}Request`
