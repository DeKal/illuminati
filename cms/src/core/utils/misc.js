import { POSTS_PER_PAGE } from 'core/const/posts'

export const isPathMatchUrl = (pathName, url, isExact = false) => {
  if (url === '/' || pathName === '/' || isExact) {
    return pathName === url
  }
  return pathName.includes(url)
}

export const stringtifyTags = (tags, limits = 20) => {
  const tagsStr = tags.join(', ')
  if (tagsStr.length > limits) {
    return tagsStr.slice(0, limits) + '...'
  }
  return tagsStr
}

export const calcMaxRowsPerPage = postCount => {
  const postsPerPageOptions = []
  for (let index = 0; index < POSTS_PER_PAGE.length; index++) {
    postsPerPageOptions.push(POSTS_PER_PAGE[index])
    if (POSTS_PER_PAGE[index] > postCount) {
      break
    }
  }
  return postsPerPageOptions
}

export const arrayToAttributes = array =>
  array.reduce((object, attribute) => {
    return {
      ...object,
      [attribute]: true
    }
  }, {})

export const attributesToArray = obj =>
  Object.keys(obj).map(attribute => (obj[attribute] ? attribute : null))
