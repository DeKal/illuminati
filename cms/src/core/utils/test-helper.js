import React from 'react'
import { shallow, mount } from 'enzyme'
import Theme from 'core/theme/ThemeWrapper'

export const shallowWithTheme = component => {
  const wrapper = shallow(
    <Theme>
      <div data-test-id="theme-wrapper">{component}</div>
    </Theme>
  )
  return wrapper.find('[data-test-id="theme-wrapper"]').shallow()
}

export const mountWithTheme = component => {
  const wrapper = mount(
    <Theme>
      <div data-test-id="theme-wrapper">{component}</div>
    </Theme>
  )
  return wrapper.find('[data-test-id="theme-wrapper"]')
}
