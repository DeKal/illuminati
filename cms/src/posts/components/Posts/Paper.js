import { withStyles } from '@material-ui/core/styles'
import Paper from '@material-ui/core/Paper'

const StylePaper = withStyles(theme => ({
  root: {
    width: '100%',
    marginBottom: theme.spacing(2)
  }
}))(Paper)

export default StylePaper
