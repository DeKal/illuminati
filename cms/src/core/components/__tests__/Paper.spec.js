import * as React from 'react'
import { mountWithTheme } from 'core/utils/test-helper'
import Paper from '../Paper'

describe('Paper', () => {
  it('should render with no error', function() {
    const wrap = mountWithTheme(<Paper />)
    expect(wrap).toMatchSnapshot()
  })
})
