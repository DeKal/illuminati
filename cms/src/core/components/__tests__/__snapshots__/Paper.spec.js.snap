// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paper should render with no error 1`] = `
<div
  data-test-id="theme-wrapper"
>
  <WithStyles(WithStyles(ForwardRef(Paper)))>
    <WithStyles(ForwardRef(Paper))
      classes={
        Object {
          "root": "WithStyles(ForwardRef(Paper))-root-1",
        }
      }
    >
      <ForwardRef(Paper)
        classes={
          Object {
            "elevation0": "MuiPaper-elevation0",
            "elevation1": "MuiPaper-elevation1",
            "elevation10": "MuiPaper-elevation10",
            "elevation11": "MuiPaper-elevation11",
            "elevation12": "MuiPaper-elevation12",
            "elevation13": "MuiPaper-elevation13",
            "elevation14": "MuiPaper-elevation14",
            "elevation15": "MuiPaper-elevation15",
            "elevation16": "MuiPaper-elevation16",
            "elevation17": "MuiPaper-elevation17",
            "elevation18": "MuiPaper-elevation18",
            "elevation19": "MuiPaper-elevation19",
            "elevation2": "MuiPaper-elevation2",
            "elevation20": "MuiPaper-elevation20",
            "elevation21": "MuiPaper-elevation21",
            "elevation22": "MuiPaper-elevation22",
            "elevation23": "MuiPaper-elevation23",
            "elevation24": "MuiPaper-elevation24",
            "elevation3": "MuiPaper-elevation3",
            "elevation4": "MuiPaper-elevation4",
            "elevation5": "MuiPaper-elevation5",
            "elevation6": "MuiPaper-elevation6",
            "elevation7": "MuiPaper-elevation7",
            "elevation8": "MuiPaper-elevation8",
            "elevation9": "MuiPaper-elevation9",
            "outlined": "MuiPaper-outlined",
            "root": "MuiPaper-root WithStyles(ForwardRef(Paper))-root-1",
            "rounded": "MuiPaper-rounded",
          }
        }
      >
        <div
          className="MuiPaper-root WithStyles(ForwardRef(Paper))-root-1 MuiPaper-elevation1 MuiPaper-rounded"
        />
      </ForwardRef(Paper)>
    </WithStyles(ForwardRef(Paper))>
  </WithStyles(WithStyles(ForwardRef(Paper)))>
</div>
`;
